HOW_TO_ADD_REST_RPC_API.md

```diff
@@ -0,0 +1,157 @@
+    }
+    return nil
+}
+```
+
+
+## Add the URL for REST RPC API
+</br>
+
+Now, we should add a URL for REST RPC API.
+
+For example, to add a REST API to generate login token for video meeting, you may want to add a RPC API named 'get_token', which is a member of the 'user' group, the REST URL can be '/user/rtc/get_token'.
+
+This URL should be add to NewChatRoute() method in 'internal/api/router.go'.
+
+```go
+func NewChatRoute(router gin.IRouter, discov discoveryregistry.SvcDiscoveryRegistry) {
+    ...
+
+	user := router.Group("/user", mw.CheckToken)
+	user.POST("/update", chat.UpdateUserInfo)                 // Edit personal information
+	user.POST("/find/public", chat.FindUserPublicInfo)        // Get user's public information
+	user.POST("/find/full", chat.FindUserFullInfo)            // Get all information of the user
+	user.POST("/search/full", chat.SearchUserFullInfo)        // Search user's public information
+	user.POST("/search/public", chat.SearchUserPublicInfo)    // Search all information of the user
+
+    // your added code begins here
+	user.POST("/rtc/get_token", chat.GetTokenForVideoMeeting) // Get token for video meeting
+
+    ...
+}
+```
+
+## Implement the REST service
+
+### Implement the REST Service Api
+
+We add REST RPC API implementation to the corresponding service implementation go file located in '/internal/api/'. For the chat service, we add codes to '/internal/api/chat.go'.
+
+```go
+...
+func (o *ChatApi) GetTokenForVideoMeeting(c *gin.Context)
+{
+	a2r.Call(chat.ChatClient.GetTokenForVideoMeeting, o.chatClient, c)
+}
+...
+```
+
+### Implement the REST Service logic
+
+We implement the REST Service logic in go files of the path '/internal/rpc/service_name/group_name.go'. For the user group of <font color="#FF8000">chat</font> service, the implementation logic should be added to '/internal/rpc/chat/user.go'.
+
+Here we call the GetLiveKitServerUrl() and GetLiveKitToken() func in the rtc package to allocate a new GetTokenForVideoMeetingResp message and return it to the RPC caller.
+
+```go
+package chat
+
+import (
+    ...
+    "github.com/OpenIMSDK/chat/pkg/common/rtc"
+    ...
+)
+
+...
+
+func (o *chatSvr) GetTokenForVideoMeeting(ctx context.Context, req *chat.GetTokenForVideoMeetingReq) (*chat.GetTokenForVideoMeetingResp, error)
+{
+    if _, _, err := mctx.Check(ctx); err != nil {
+        return nil, err
+    }
+    serverUrl := rtc.GetLiveKitServerUrl()
+    token, err := rtc.GetLiveKitToken(req.Room, req.Identity)
+    if err != nil {
+        return nil, err
+    }
+    return &chat.GetTokenForVideoMeetingResp{
+        ServerUrl: serverUrl,
+        Token:     token,
+    }, err
+}
+```
+
+## Done
+
+Now we have finished all the works to add an REST RPC API for OpenIM Chat. We may call
+
+```shell
+make build
+```
+to re-compile the chat project and call
+
+```shell
+make start
+```
+to start the OpenIM Chat server and test the <font color="#FF8000">chat</font> service's new REST RPC API.
+
+### 
```

Comment:
why /

Reply:
To explain how to add a REST RPC API for the OpenIM Chat, demanded by sk.